src/prototypecomponentsnavigatorwindow.cpp

```diff
@@ -0,0 +1,79 @@
+#include "prototypecomponentsnavigatorwindow.h"
+#include "ui_prototypecomponentsnavigatorwindow.h"
+#include <QItemSelectionModel>
+#include <QAbstractItemModel>
+#include <QStandardItem>
+
+PrototypeComponentsNavigatorWindow::PrototypeComponentsNavigatorWindow(QWidget *parent) :
+    QWidget(parent),
+    ui(new Ui::PrototypeComponentsNavigatorWindow)
+{
+    ui->setupUi(this);
+
+
+    mSelectionModel = new QItemSelectionModel;
+
+    ui->listView->setIconSize(QSize(100,100));
+    ui->listView->setUniformItemSizes(true);
+    ui->listView->setSelectionMode(QAbstractItemView::MultiSelection);
+
+    connect(ui->pushButtonAddComponentsToPrototipe, &QPushButton::clicked, this, &PrototypeComponentsNavigatorWindow::newComponentButtonClicked);
+    connect (ui->pushButtonLoadSelectedComponents, &QPushButton::clicked, this, &PrototypeComponentsNavigatorWindow::componentOpenButtonClicked);
+    connect(ui->pushButtonDeleteSelectedComponent, &QPushButton::clicked, this, &PrototypeComponentsNavigatorWindow::componentRemovButtonClicked);
+
+}
+
+PrototypeComponentsNavigatorWindow::~PrototypeComponentsNavigatorWindow()
+{
+    delete ui;
+}
+
+void PrototypeComponentsNavigatorWindow::setModel(QAbstractItemModel *model)
+{
+    if(model)
+    {
+        mPreviewItemModel = model;
```

Comment:
Do you really need to store the item model in a separate class member? You always have access to your view, so you can take the selection model there every time you need it.